Sources/PackageLoading/TargetSourcesBuilder.swift

```diff
@@ -83,12 +83,39 @@ public struct TargetSourcesBuilder {
             }
         }
         self.declaredSources = declaredSources?.spm_uniqueElements()
+        
+        for exclude in self.excludedPaths {
+            validTargetPath(exclude, "Cannot exclude path: \(exclude) because it does not exist")
+        }
+        
+        if let decSources = self.declaredSources {
+            for source in decSources {
+                validTargetPath(source, "Source: \(source) does not exist")
+            }
+        }
 
       #if DEBUG
         validateRules(self.rules)
       #endif
     }
 
+    @discardableResult
+    private func validTargetPath(_ absPath: AbsolutePath, _ message: String) -> Bool {
```

Comment:
a more "swifty" signature would be `validTargetPath(at: AbsolutePath, message: String?)` and only emit the message if set